libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -309,7 +309,15 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
 
     // loop through groups of 4 ESCs
     const uint8_t esc_offset = constrain_int16(mavlink_offset, 0, ESC_TELEM_MAX_ESCS-1);
-    const uint8_t num_idx = ESC_TELEM_MAX_ESCS/4;
```

Comment:
Truth Table:
```
MAX_ESC -> num_idx
0 -> 0
1 -> 1
2 -> 1
3 -> 1
4 -> 1
5 -> 2
6 -> 2
7 -> 2
8 -> 2
9 -> 3
10 -> 3
```
What we really want is to always send at least one so it's sort of a round-up type of thing. How about this logic?
`const uint8_t num_idx = (ESC_TELEM_MAX_ESCS/4) + 1;`

That logic gives a wrong result for ESC_TELEM_MAX_ESCS == 0 but that probably breaks other things too which is why there's a [static_assert](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_ESC_Telem/AP_ESC_Telem.h#L11) to check for that.

Reply:
Can you please help me with the functional meaning of ESC_TELEM_MAX_ESCS == 0 ?
IF this is zero then what it means because I have handled that if it is 0 then it is divisible by 4 and as it is 0 it should not go into the loop 
Means loop will start only if at least 1 is ESC_TELEM_MAX is present

Reply:
I have rectified the variable declaration in commit with sha 80d22a4c4caf7fcc1a96375cb82ff8c7da78bf7d  @magicrub

Reply:
==0 won't compile. Don't worry about it.